src/OUNITAT.cpp

```diff
@@ -555,7 +555,7 @@ void Unit::hit_firm(Unit* attackUnit, int targetXLoc, int targetYLoc, float atta
 	if(!locPtr->is_firm())
 		return;	// do nothing if no firm there
 
-	Nation *attackNation = nation_array[attackNationRecno];
+	Nation *attackNation = attackNationRecno ? nation_array[attackNationRecno] : nullptr;
```

Comment:
But DynArray[0] already returns null. Did you actually hit a crash? Please send a stack trace.

Reply:
Yes I actually hit a crash, which is why I found it. The attackNationRecno can be 0 when the attack is initiated from a rebel. Please merge the crash-fix for OUNITAT.

Reply:
You are compiling with DEBUG (custom assertion code) turned on. It's not a real crash though. Let me think about this.

Reply:
Yeah, the `DEBUG` macro helps find problems early. The original code didn't trigger the assertion in `operator[]` and neither should the modification - all (other) code follows this pattern of checking before accessing, which is the correct thing to do. It's also better to be explicit about the ability for `attackNation` to be null, because else readers of the code will tacitly assume that it has a value.